docs/src/installation_upgrade.md

```diff
@@ -236,6 +236,150 @@ When versions are not directly upgradable, the old version needs to be
 removed before installing the new one. This won't affect user data but
 only the operator itself.
 
+### Upgrading to 1.21 from a previous minor version
+
+!!! Important
+    If you are upgrading from 1.19 or older, please read the
+    [instructions to upgrade to version 1.20](#upgrading-to-120-from-a-previous-minor-version)
+    first.
+
+With the goal to keep improving out-of-the-box the *convention over
+configuration* behavior of the operator, CloudNativePG 1.21 changes the default
+value of several knobs in the following areas:
+
+- startup and shutdown control of the PostgreSQL instance
+- self-healing
+- security
+- labels
+
+Most of the changes will affect new PostgreSQL clusters only.
+
+!!! Warning
+    Please read carefully the list of changes below and how to change the
+    `Cluster` manifests to retain the existing behavior, in case you don't want to
+    disrupt your existing workloads. Alternatively, postpone the upgrade to 1.21
+    until you are sure. In general, we recommend adopting these default
+    values unless you have valid reasons not to.
+
+If you want to keep the existing behavior of CloudNativePG explicitly,
+preferably just temporarily, you need to set these values in all your `Cluster`
+definitions **before upgrading** to version 1.21:
+
+```yaml
+spec:
+   ...
+   startDelay: 30
+   stopDelay: 30
+   switchoverDelay: 40000000
+   enableSuperuserAccess: true
+   replicationSlots:
+     highAvailability:
+       enabled: false
+```
+
+#### Delay for PostgreSQL shutdown
+
+Up to now, [the `stopDelay` parameter](instance_manager.md#shutdown-control)
+was set to 30 seconds. Despite the recommendations to change and tune this
+value, almost all the cases we have examined during support incidents or
+community issues show that this value is left unchanged.
+
+The [new value is set to 1800 seconds](https://github.com/cloudnative-pg/cloudnative-pg/commit/9f7f18c5b9d9103423a53d180c0e2f2189e71c3c),
+the equivalent of 30 minutes.
+
+A new parameter has been introduced to define the maximum time window within
+the `stopDelay` one, reserved to gracefully stop PostgreSQL using the `smart`
+shutdown procedure. Once completed, the remaining time up to `stopDelay` will
+be reserved for PostgreSQL to complete its duties regarding WAL commitments
+with both the archive and the streaming replicas to ensure the cluster
+doesn't lose any data.
+
+If you want to retain the old behavior, you need to set explicitly:
+
+```yaml
+spec:
+   ...
+   stopDelay: 30
+```
```

Comment:
AFAIU, we cannot achieve the previous behavior with the current code. Setting stopDelay to 30 will leave the smartStopDelay set to 30 too, which means we are not giving any time to Postgres to run the shutdown fast before killing the pod (besides the 2-second grace period gifted by the kubelet). I do not know if that's intentional, but I think we should at least write it in the documentation.

Reply:
What we can do then is change the minimum slot to 15 seconds instead of 30 in the formula that calculates the minimum time.